src/components/TokenBalance.test.tsx

```diff
@@ -54,14 +54,357 @@ jest.mocked(getDynamicConfigParams).mockReturnValue({
+          "title": "outOfSyncBanner.title",
+          "type": "ALERT/SHOW",
+          "underlyingError": undefined,
+        },
+      ]
+    `)
+  })
+
+  it('renders correctly when hideBalance is true', async () => {
+    const store = createMockStore({ ...defaultStore, app: { hideHomeBalances: true } })
+
+    const tree = render(
+      <Provider store={store}>
+        <HomeTokenBalance />
+      </Provider>
+    )
+
+    expect(getElementText(tree.getByTestId('TotalTokenBalance'))).toEqual('XX.XX')
+    expect(tree.getByTestId('HiddenEyeIcon')).toBeTruthy()
+  })
+
+  it('renders correctly when hideBalance is false', async () => {
+    const store = createMockStore(defaultStore)
+
+    const tree = render(
+      <Provider store={store}>
+        <HomeTokenBalance />
+      </Provider>
+    )
+
+    expect(getElementText(tree.getByTestId('TotalTokenBalance'))).toEqual('$8.41')
+    expect(tree.getByTestId('EyeIcon')).toBeTruthy()
+  })
+
+  it('renders correctly when feature flag is off, hideBalance is false', async () => {
+    jest
+      .mocked(getFeatureGate)
+      .mockImplementation(
+        (featureGate) => featureGate !== StatsigFeatureGates.SHOW_HIDE_HOME_BALANCES_TOGGLE
+      )
+    const store = createMockStore(defaultStore)
+
+    const tree = render(
+      <Provider store={store}>
+        <HomeTokenBalance />
+      </Provider>
+    )
+
+    expect(getElementText(tree.getByTestId('TotalTokenBalance'))).toEqual('$8.41')
+    expect(tree.queryByTestId('EyeIcon')).toBeFalsy()
+    expect(tree.queryByTestId('HiddenEyeIcon')).toBeFalsy()
+  })
+
+  it('renders correctly when feature flag is off, hideBalance is true', async () => {
     jest
       .mocked(getFeatureGate)
       .mockImplementation(
-        (featureGate) => featureGate !== StatsigFeatureGates.SHOW_ASSET_DETAILS_SCREEN
+        (featureGate) => featureGate !== StatsigFeatureGates.SHOW_HIDE_HOME_BALANCES_TOGGLE
       )
+    const store = createMockStore({ ...defaultStore, app: { hideHomeBalances: true } })
+
+    const tree = render(
+      <Provider store={store}>
+        <HomeTokenBalance />
+      </Provider>
+    )
+
+    expect(getElementText(tree.getByTestId('TotalTokenBalance'))).toEqual('$8.41')
+    expect(tree.queryByTestId('EyeIcon')).toBeFalsy()
+    expect(tree.queryByTestId('HiddenEyeIcon')).toBeFalsy()
+  })
+
+  it('tracks analytics event when eye icon is pressed', async () => {
+    const store = createMockStore(defaultStore)
+
+    const tree = render(
+      <Provider store={store}>
+        <HomeTokenBalance />
+      </Provider>
+    )
+    fireEvent.press(tree.getByTestId('EyeIcon'))
+    expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+    expect(ValoraAnalytics.track).toHaveBeenCalledWith(HomeEvents.hide_balances)
+  })
+})
+
+describe('FiatExchangeTokenBalance and HomeTokenBalance', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.mocked(getFeatureGate).mockReturnValue(true)
```

Comment:
can drop this too since this doesn't do any tests related to the balance toggle
feature gate

Reply:
Was able to drop this feature gate in fcedcf6c96c47fbb27854528ba7e1a59633efdb3.